core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/waiters/DefaultAsyncWaiter.java

```diff
@@ -36,16 +36,16 @@
 @SdkInternalApi
 @ThreadSafe
 public final class DefaultAsyncWaiter<T> implements AsyncWaiter<T> {
-    private final PollingStrategy pollingStrategy;
+    private final WaiterOverrideConfiguration overrideConfiguration;
     private final ScheduledExecutorService executorService;
     private final List<WaiterAcceptor<? super T>> waiterAcceptors;
     private final AsyncWaiterExecutor<T> handler;
 
     private DefaultAsyncWaiter(DefaultBuilder<T> builder) {
         this.executorService = builder.scheduledExecutorService;
-        this.pollingStrategy = Validate.paramNotNull(builder.pollingStrategy, "pollingStrategy");
+        this.overrideConfiguration = Validate.paramNotNull(builder.overrideConfiguration, "overrideConfiguration");
```

Comment:
Is it really required? Can we create one with default values? Doesn't feel like override config if it MUST be set.

Reply:
Yeah, good catch. Will update